File: backend/internal/server/handler/buyer/response/purchase.go
package response

// Purchase represents a view of a purchase for the buyer.
type Purchase struct {
	ID          int    `json:"id"`
	ItemID      int    `json:"item_id"`
	FishType    string `json:"fish_type"`
	Quantity    int    `json:"quantity"`
	Unit        string `json:"unit"`
	Price       int    `json:"price"`
	BuyerID     int    `json:"buyer_id"`
	AuctionID   int    `json:"auction_id"`
	AuctionDate string `json:"auction_date"`
	CreatedAt   string `json:"created_at"`
}

File: frontend/src/features/mypage/types/purchase.ts
import { Purchase as EntityPurchase } from '@/src/data/api/buyer_mypage';

export interface Purchase {
  id: number;
  itemId: number;
  createdAt: string;
  fishType: string;
  quantity: number;
  unit: string;
  auctionId: number;
  auctionDate: string;
  price: number;
}

export const toPurchase = (entity: EntityPurchase): Purchase => ({
  id: entity.id,
  itemId: entity.itemId,
  fishType: entity.fishType,
  quantity: entity.quantity,
  unit: entity.unit,
  price: entity.price,
  auctionId: entity.auctionId,
  auctionDate: entity.auctionDate,
  createdAt: entity.createdAt,
});
